### PR TITLE
[Java] Add isDetached Option

### PR DESCRIPTION
**Pull request by lixin-wei**
## Why are these changes needed?

Add isDetached actor creation option to Java API.

## Related issue number

https://github.com/ray-project/ray/issues/20353
## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by jovany-wang — APPROVED** [review]
LGTM

**Review by kfstorm — COMMENTED** [review]
LGTM overall, but we need to add a unit test.

**Comment by kfstorm** [comment]
BTW, in Python we use `MyActor.options(lifetime="detached").remote()` to create a detached actor. Maybe we need to use a lifetime enum in Java as well.

**Comment by jovany-wang** [comment]
> BTW, in Python we use `MyActor.options(lifetime="detached").remote()` to create a detached actor. Maybe we need to use a lifetime enum in Java as well.

Yes. But the value of the enum should be detached or extends from the parent right?

**Comment by kfstorm** [comment]
@jovany-wang What do you mean by "extends from the parent"?

**Comment by jovany-wang** [comment]
> @jovany-wang What do you mean by "extends from the parent"?

Sorry. I think what I'd like to mean is fate sharing with the parent actor.

**Comment by kfstorm** [comment]
@jovany-wang We can just create an enum `ActorLifetime` with `Default` and `Detached` options. We can add some comments on the options about the detailed behaviors.

**Comment by jovany-wang** [comment]
> @jovany-wang We can just create an enum `ActorLifetime` with `Default` and `Detached` options. We can add some comments on the options about the detailed behaviors.

I see.
@lixin-wei Do you intended to introduce ActorLifetime enum and add unit test for it?

File: java/api/src/main/java/io/ray/api/call/BaseActorCreator.java [hunk 1]
```diff
@@ -85,6 +85,18 @@ public T setMaxConcurrency(int maxConcurrency) {
...
   }
 
+  /**
+   * Set whether this actor is detached.
+   *
+   * @param isDetached Whether this actor is detached.
+   * @return self
+   * @see ActorCreationOptions.Builder#isDetached(boolean)
+   */
+  public T isDetached(boolean isDetached) {
```
Comment (raulchen) [inline java/api/src/main/java/io/ray/api/call/BaseActorCreator.java:95]: better name this `setDetached`. `isDetached` sounds like a getter method.

**Review by raulchen — COMMENTED** [review]
please add a test

**Review by raulchen — CHANGES_REQUESTED** [review]
Right, to be aligned with Python, we should use `setLifetime(ActorLifetime.DETACHED)`. 
Also, please change the `actor.rst` doc file as well.

**Pull request closed without merging**